frontend/test/metabase/scenarios/dashboard/action-permissions.cy.spec.js

```diff
@@ -0,0 +1,56 @@
+import { onlyOn } from "@cypress/skip-test";
+import { restore, popover } from "__support__/cypress";
+
+// TODO Add other permissions
+const PERMISSIONS = {
+  curate: ["admin", "normal"],
+  view: ["readonly"],
+};
+
+describe("dashboard action permissions", () => {
+  beforeEach(() => {
+    restore();
+  });
+
+  Object.entries(PERMISSIONS).forEach(([permission, userGroup]) => {
+    context(`${permission} access`, () => {
+      userGroup.forEach(user => {
+        beforeEach(() => {
+          cy.signIn(user);
+          cy.visit("/dashboard/1");
+        });
+
+        describe(`${user} user`, () => {
+          onlyOn(permission === "curate", () => {
+            it("should be able to duplicate dashboard", () => {
+              ellipsisMenuPopover().within(() => {
+                cy.findByText("Duplicate").click();
+              });
```

Comment:
When there's only one string, this can also work:
```javascript
ellipsisMenuPopover()
  .findByText("Duplicate")
  .click();
```

Btw, nice abstraction. I never thought of doing some assertions first in a function and then returning an element to be able to chain other `cy` commands off of it.

Reply:
Thank, it's much more clean now. Glad you like the approach 🤝